### ISSUE TITLE
DRE: Wrap in-dialogue stage directions (modern text) in brackets

### ISSUE THREAD

**Issue by JackWilton1594** [projectLEMDO/lemdoIssues#233]
When stage directions appear in the dialogue of modern texts, they should be rendered in italics and wrapped in brackets.

Here are two examples from the opening scene of 1HW:

![Screen Shot 2024-06-06 at 15 31 46](https://github.com/projectLEMDO/lemdoIssues/assets/4774420/ae8b6e0f-6705-4343-809c-291e614c1db9)

These should render as: 
- _(To the Duke)_ I do arrest thee, murderer.
- _(To Attendants)_ Set down,

The guidelines ask editors _not_ to add parentheses or italics in their texts because they will be rendered this way by the system.

It's important that the brackets that wrap SDD are themselves rendered in italics – that way, they will be distinguishable from parentheses in dialogue.

**Comment by martindholmes** [comment]
I've accomplished this through CSS in rev 18085. This adds CSS parentheses around stage directions which are tagged as delivery. In the results, I see some already have hard-coded parentheses in 1HW, so they now show as doubled. @JackWilton1594 Can you check to see if the results are what you expected?

**Comment by JackWilton1594** [comment]
Thanks, Martin. It isn't only delivery SDD that need wrapping. It's any SDD embedded in dialogue rather than appearing on their own line. 

I appreciate we'll have to go through and remove hard-coded parentheses where they are in the text.

Of course, this is just for DRE/NISE – other anthologies may wish to continue with practice of using square brackets. We just don't think SDD require extra signalling in this way when we don't use square brackets in dialogue.

**Comment by JanelleJenstad** [comment]
QME will need some kind of override because they wanted to keep their square brackets around child `<supplied>` elements in `<stage>`. Our processing resulting in square brackets and round parentheses, which they didn't like. I know we sorted it out for their release in the fall. We just need to check that these new changes don't have any fall-out for QME's beta site.

**Comment by JackWilton1594** [comment]
Confusingly, there are double parentheses in some SDD. Here in AHDM, for example:

![Screen Shot 2024-06-12 at 23 56 15](https://github.com/projectLEMDO/lemdoIssues/assets/4774420/7abd24b7-0a53-46f6-a84f-12ef69c5a500)

There aren't any hard-coded parentheses in the XML:

`<sp who="#emdAHDM_M_Lemot" xml:id="emdAHDM_M_s2_sp17">
               <speaker>Lemot</speaker>
               <p>
                  <stage type="delivery" xml:id="emdAHDM_M_s2_sd3"><anchor xml:id="emdAHDM_M_anc_136"/>Aside to Colinet<anchor xml:id="emdAHDM_M_anc_137"/></stage> See where he comes. Now must I say, <foreign xml:lang="la"><anchor xml:id="emdAHDM_M_anc_138"/>Lupus est in fabula<anchor xml:id="emdAHDM_M_anc_139"/></foreign>, for these Latin ends are part of a gentleman and a good scholar.</p>
            </sp>`

**Comment by martindholmes** [comment]
Maybe we already had a rule to distinguish asides by using parentheses; adding them to non-asides would remove that distinction, so I suspect we need to think again. This needs a decision by everyone on what the various types of stage directions should look like. I'll remove the change I made in rev 18085 and wait for a consensus and a detailed outline of the various stage direction types, contexts, and display rules. We can worry about QME later.

**Comment by martindholmes** [comment]
Reversed in rev 18266. This ticket's now waiting on a taxonomy of stage-direction types and contexts, with guidelines on display for each type/context.

**Comment by JanelleJenstad** [comment]
@JackWilton1594: We'll need to meet to decide on further instructions for Martin.

**Comment by JackWilton1594** [comment]
@JanelleJenstad Sounds good. Apologies for only catching up on these tickets now!

**Comment by JackWilton1594** [comment]
@JanelleJenstad I don't think we covered this topic in our meeting. My sense is that in-line directions -- that is, @stage within @p or @l -- should inherit the alignment of the line/paragraph and rendered _(enclosed within italicised parentheses)_. 

Any directions that require their own line -- e.g. entrances and exits -- shouldn't be included in @p or @l

**Comment by JanelleJenstad** [comment]
@JackWilton1594: I agree that entrances and exits shouldn't be included in `<p>` or `<l>` elements ... unless it's a character exiting during someone else's speech, as can happen when a character dispatches a servant, for example. As we proofread during the current pre-freeze period, we'll need to proofread the encoding to make sure that `<stage>` is where we want it to be. Many entrances and exits came over from IML inside speeches. Bottom line: We're responsible for the encoding, so let's always check that first.

**Comment by JanelleJenstad** [comment]
@martindholmes: We have achieved consensus. I agree with Brett that **in-line directions -- that is, `<stage>` within `<p>` or `<l>` -- should inherit the alignment of the line/paragraph and be rendered inside parentheses.** Both the parenthesis marks and the content should be italicized.

**Comment by martindholmes** [comment]
Do we now need Schematron to prevent entrance and exit sds in speeches?

**Comment by JanelleJenstad** [comment]
@martindholmes: Your reversal in June fixed the immediate problem.  
![image](https://github.com/user-attachments/assets/2f9937d7-4f68-41c2-87d3-8cb78d51494a)

**Comment by JanelleJenstad** [comment]
> Do we now need Schematron to prevent entrance and exit sds in speeches?

No, because other characters can enter and exit during speeches. What we're looking to avoid is exits and entrances within `<p>` or `<l>` inside speeches. 

Good encoding:
`<sp>
<speaker></speaker>
<l>Begone, foul fiend!</l>
<stage type="exit">Exit Fiend</stage>
<l>Whew! Thank God he's gone.</l>
</sp>`

Not good:
`<sp>
<speaker></speaker>
<l>Begone, foul fiend!<stage type="exit">Exit Fiend</stage></l>
<l>Whew! Thank God he's gone.</l>
</sp>`

**Comment by JanelleJenstad** [comment]
I've created new ticket #289 to deal with a taxonomy of SD (if it's necessary) and new ticket #290 for Schematron. Closing this ticket.

**Comment by JanelleJenstad** [comment]
> Do we now need Schematron to prevent entrance and exit sds in speeches?

No. We need to allow entrances and exits within speeches.

**Comment by JanelleJenstad** [comment]
> > Do we now need Schematron to prevent entrance and exit sds in speeches?
> 
> No, because other characters can enter and exit during speeches. What we're looking to avoid is exits and entrances within `<p>` or `<l>` inside speeches.
> 
> Good encoding: `<sp> <speaker></speaker> <l>Begone, foul fiend!</l> <stage type="exit">Exit Fiend</stage> <l>Whew! Thank God he's gone.</l> </sp>`
> 
> Not good: `<sp> <speaker></speaker> <l>Begone, foul fiend!<stage type="exit">Exit Fiend</stage></l> <l>Whew! Thank God he's gone.</l> </sp>`

Unless the exit or entrance is inline.
